sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionState.scala

```diff
@@ -29,7 +29,7 @@ import org.apache.spark.sql.internal.{SessionState, SQLConf}
 /**
  * A class that holds all session-specific state in a given [[HiveContext]].
  */
-private[hive] class HiveSessionState(ctx: HiveContext) extends SessionState(ctx) {
+private[hive] class HiveSessionState(val ctx: HiveContext) extends SessionState(ctx) {
```

Comment:
please don't do this; if someone needs the context then we should explicitly pass in the context itself instead of getting it through the session state.